### PR TITLE
Update to 0.6.1

### PR DESCRIPTION
**Pull request by sumit0190**
This is required for building `tensorboard` for `osx-arm64`. We could also just rebuild `0.6` for `osx-arm64`, but upgrading is probably the better choice. This also needs the latest `rust` compiler.

### CODE REVIEW

**Comment by sumit0190** [comment]
The `bazel` rules in `tensorboard` need to be updated for `osx-arm64` anyway, so just updating `tensorboard-data-server` won't amount to much. Closing this PR for now and I'll revisit this when `osx-arm64` support for `tensorboard` is a bigger priority.

**Review by varlackc — COMMENTED** [review]
we probably might want to add `patch` for platforms other than `windows` as the following example:

`patch  # [not win]`

https://github.com/AnacondaRecipes/tensorboard-data-server-feedstock/blob/3b2127f2bf2063406b482ffd44dff774cbec2088/recipe/meta.yaml#L26&L28

We might want to add a `doc_url` and a `description`
https://github.com/AnacondaRecipes/tensorboard-data-server-feedstock/blob/3b2127f2bf2063406b482ffd44dff774cbec2088/recipe/meta.yaml#L42&L47

maybe the following url might be useful for the `doc_url` but we might need to do some research:
https://www.tensorflow.org/tensorboard/get_started

**Review by cbouss — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,17 +23,14 @@ build:
...
+    - {{ compiler('rust') }}            # [not win]
+    - m2-patch                          # [win]
   host:
     - python
     - pip
-    - openssl  # [linux]
-    - libiconv # [osx and x86_64]
+    - openssl                           # [not (win or osx or linux-64)]
   run:
     - python
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:37]: Would be nice to have a `pip check` command

File: recipe/meta.yaml [hunk 2]
```diff
@@ -23,17 +23,14 @@ build:
 
 requirements:
   build:
-    - git
-    - patch       # [unix]
-    - m2-patch    # [win]
-    - python                                 # [build_platform != target_platform]
-    - {{ compiler('c') }}  # [not win]
-    - {{ compiler('rust') }}  # [not win]
+    - python                            # [build_platform != target_platform]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: As we don't build cross-platform you can remove `python                            # [build_platform != target_platform]` from the `build` section

File: recipe/meta.yaml [hunk 3]
```diff
@@ -23,17 +23,14 @@ build:
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:24]: Skip `py<36`, https://github.com/tensorflow/tensorboard/blob/6acf0be88b5727e546dd64a8b9b12d790601d561/tensorboard/data/server/pip_package/setup.py#L35

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**